components/renku_data_services/errors/errors.py

```diff
@@ -45,6 +45,16 @@ class UnauthorizedError(BaseError):
     quiet: bool = True
 
 
+@dataclass
+class InvalidOAuth2Token(BaseError):
```

Comment:
Why not have `class InvalidOAuth2Token(UnauthorizedError):` and only override the `message` field?